network-api/networkapi/wagtailpages/tests/buyersguide/test_forms.py

```diff
@@ -0,0 +1,117 @@
+            {
+                **data,
+                "title": "Test Article Title",
+                "slug": "test-article-title",
+                "body": streamfield([("paragraph", rich_text("Hello world!"))]),
+            }
+        )
+
+    def test_article_page_requires_search_image(self):
+        """
+        Test that the buyersguide article page form will
+        return a validation error if no "search_image" is set.
+        """
+        test_form = self.article_page_form(
+            data=self.generate_form_data(
+                {
+                    "search_description": "Test search description",
+                    "search_image": None,
+                },
+            ),
+        )
+
+        self.assertFalse(test_form.is_valid())
+        self.assertEqual(1, len(test_form.errors))
+        self.assertIn("search_image", test_form.errors)
+        self.assertIn("This field is required.", test_form.errors["search_image"])
+
+    def test_article_page_requires_search_description(self):
+        """
+        Test that the buyersguide article page form will
+        return a validation error if no "search_description" is set.
+        """
+        article_page = buyersguide_factories.BuyersGuideArticlePageFactory()
+
+        test_form = self.article_page_form(
+            data=self.generate_form_data(
+                {
+                    "search_description": None,
+                    "search_image": article_page.search_image,
+                },
+            ),
+        )
+
+        self.assertFalse(test_form.is_valid())
+        self.assertEqual(1, len(test_form.errors))
+        self.assertIn("search_description", test_form.errors)
+        self.assertIn("This field is required.", test_form.errors["search_description"])
+
+    def test_article_page_requires_both_search_fields(self):
+        """
+        Test that the buyersguide article page form will return validation
+        errors for both "search_image" and "search_description" fields
+        if neither are updated.
+        """
+        test_form = self.article_page_form(
+            data=self.generate_form_data(
+                {
+                    "search_description": None,
+                    "search_image": None,
+                },
+            ),
+        )
+
+        self.assertFalse(test_form.is_valid())
+        self.assertEqual(2, len(test_form.errors))
+        self.assertIn("search_image", test_form.errors)
+        self.assertIn("search_description", test_form.errors)
+        self.assertIn("This field is required.", test_form.errors["search_image"])
+        self.assertIn("This field is required.", test_form.errors["search_description"])
+
+    def test_article_page_with_search_fields_is_valid(self):
+        """
+        Test that a buyersguide article page form with
+        the search fields set is valid with no errors.
+        """
+        article_page = buyersguide_factories.BuyersGuideArticlePageFactory()
```

Comment:
Basically the same question as above about why we need to generate an article page?